analyses/country_template/docs/trigger_report_template.Rmd

```diff
@@ -0,0 +1,333 @@
+lapply(packages, library, character.only = TRUE)
+
+```
+
+```{r functions}
+computeCIwidths <- function(metric_name, trigger_id) { 
+
+  perf_metrics_data %>%
+  filter(trigger == trigger_id) %>%
+  select(-trigger) %>%
+  filter(metric == metric_name) %>%
+  pivot_wider(names_from = point, values_from = value) %>%
+  mutate(below_ci = low_end,  # width between zero and low end of CI
+         ci = high_end - low_end,# width of CI
+         above_ci = 100 - high_end) %>% # width between high_end of CI and 100
+  pivot_longer(!c('metric','dummy'), names_to = 'point', values_to = 'value')
+}
+```
+
+```{r data loading and formatting}
+
+# read in performance metrics file
+perf_metrics_data <- read.csv(perf_metrics_filename,
+                              colClasses = c('character', 'character', 'character', 'integer', 'integer')
+                              )
+
+# format funding amounts
+funding_a <- paste0('$',funding_a,'m')
+funding_b <- paste0('$',funding_b,'m')
+funding_c <- paste0('$',funding_c,'m')
+
+```
+
+```{r compute CI measurements}
+
+# compute width and position of CIs for trigger a
+df_a_var <- computeCIwidths(metric_name = 'var', trigger_id = 'a') # valid activation rate 
+df_a_det <- computeCIwidths(metric_name = 'det', trigger_id = 'a') # detection rate
+df_a_far <- computeCIwidths(metric_name = 'far', trigger_id = 'a') # false alarm rate
+df_a_mis <- computeCIwidths(metric_name = 'mis', trigger_id = 'a') # miss rate
+df_a_atv <- computeCIwidths(metric_name = 'atv', trigger_id = 'a') # probability of trigger activation
+
+# compute width and position of CIs for trigger b
+df_b_var <- computeCIwidths(metric_name = 'var', trigger_id = 'b') # valid activation rate 
+df_b_det <- computeCIwidths(metric_name = 'det', trigger_id = 'b') # detection rate
+df_b_far <- computeCIwidths(metric_name = 'far', trigger_id = 'b') # false alarm rate
+df_b_mis <- computeCIwidths(metric_name = 'mis', trigger_id = 'b') # miss rate
+df_b_atv <- computeCIwidths(metric_name = 'atv', trigger_id = 'b') # probability of trigger activation
+
+# compute width and position of CIs for trigger c
+df_c_var <- computeCIwidths(metric_name = 'var', trigger_id = 'c') # valid activation rate 
+df_c_det <- computeCIwidths(metric_name = 'det', trigger_id = 'c') # detection rate
+df_c_far <- computeCIwidths(metric_name = 'far', trigger_id = 'c') # false alarm rate
+df_c_mis <- computeCIwidths(metric_name = 'mis', trigger_id = 'c') # miss rate
+df_c_atv <- computeCIwidths(metric_name = 'atv', trigger_id = 'c') # probability of trigger activation
+
+# compute width and position of CIs for the framework
+df_fmwk_min <- computeCIwidths(metric_name = 'min', trigger_id = 'framework') # prob minimum framework activation 
+df_fmwk_ful <- computeCIwidths(metric_name = 'ful', trigger_id = 'framework') # prob full framework activation 
+
+```
+
+```{r data plotting}
+
+source("computeCIplots.R")
+
+plotCI(trigger_id = 'a', metric = 'var', ci_widths_df = 'df_a_var') 
```

Comment:
I think you could also use `map` here to simplify

Reply:
Same as above

Reply:
Removing this change because the plotCI function doesn't handle lists well. Will look into ti when we revisit this Rmd to reflect feedback from colleagues.